src/librustc_mir/interpret/memory.rs

```diff
@@ -313,7 +316,8 @@ impl<'a, 'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'a, 'mir, 'tcx, M> {
                 Some(alloc) => Ok(alloc),
                 None => {
                     // static alloc?
-                    match self.tcx.alloc_map.lock().get(id) {
+                    let alloc = self.tcx.alloc_map.lock().get(id);
+                    match alloc {
```

Comment:
Why this change?

Reply:
because otherwise the `alloc_map` stays locked within the match arms, which would not allow the `Static` case to access it again.

Reply:
I get it